src/v/cluster/feature_backend.cc

```diff
@@ -39,7 +42,31 @@ feature_backend::apply_update(model::record_batch b) {
             });
       });
 
+    // Updates to the feature table are very infrequent, usually occurring
+    // once during an upgrade.  Snapshot on every write, so that as soon
+    // as a feature has gone live, we can expect that on subsequent startup
+    // it will be live from early in startup (as soon as storage subsystem
+    // loads and snapshot can be loaded)
+    co_await save_snapshot();
+
     co_return errc::success;
 }
 
+ss::future<> feature_backend::save_snapshot() {
+    auto snapshot = features::feature_table_snapshot::from(
+      _feature_table.local());
+
+    auto val_bytes = serde::to_iobuf(snapshot);
+    auto key_bytes = serde::to_iobuf(ss::sstring("feature_table"));
+
+    // kvstore is shard-local: must be on a consistent shard every
+    // time for snapshot storage to work.
+    vassert(ss::this_shard_id() == ss::shard_id{0}, "wrong shard");
```

Comment:
nit: put this up top? Just to give future readers the context that this is shard 0 up front

Reply:
fair, done.